src/js/utilities/truncateText.js

```diff
@@ -0,0 +1,12 @@
+
+export const truncateText = (text, maxLength) => {
+    if (text.length <= maxLength) return text;
+  
+    const truncated = text.slice(0, maxLength).trimEnd();
+    const lastSpaceIndex = truncated.lastIndexOf(' ');
+  
+    return lastSpaceIndex > 0
+      ? truncated.slice(0, lastSpaceIndex) + '...'
+      : truncated + '...';
+  };
+  
```

Comment:
Your truncateText function is really good and functional! 
A very clean and correct code!
It is great that you are using functions like slice, trimEnd() and lastIndexOf()